packages/flutter_test/lib/src/test_compat.dart

```diff
@@ -250,9 +256,27 @@ void setUpAll(dynamic Function() body) {
 /// prefer [tearDown], and only use [tearDownAll] if the callback is
 /// prohibitively slow.
 void tearDownAll(dynamic Function() body) {
+  _configureTearDownForTestFile();
   _declarer.tearDownAll(body);
 }
 
+bool _isTearDownForTestFileConfigured = false;
+/// Configures `tearDownAll` after all user defined `tearDownAll` in the test file.
+///
+/// This function should be invoked in all functions, that may be invoked by user in the test file,
+/// to be invoked before any other `tearDownAll`.
+void _configureTearDownForTestFile() {
```

Comment:
@natebosch, created hook as we discussed. Can you check it looks ok?

It is still little hacky. An alternative would be to make `tearDownForTestFile` a first citizen and put it to declarer, but it will make the method public that we maybe do not want, because end user does not need it.

What do you think?